library/aes.c

```diff
@@ -33,6 +33,30 @@
 #include "mbedtls/platform.h"
 #include "mbedtls/platform_util.h"
 #include "mbedtls/error.h"
+
+#if defined(__aarch64__)
+#if !defined(MBEDTLS_AESCE_C) && defined(MBEDTLS_AES_USE_HARDWARE_ONLY)
+#error "MBEDTLS_AES_USE_HARDWARE_ONLY defined, but not all prerequisites"
+#endif
+#endif
+
+#if defined(__amd64__) || defined(__x86_64__) || \
+    defined(_M_X64) || defined(_M_AMD64)
```

Comment:
Surprisingly, `_M_AMD64` and `_M_X64` are defined for Arm64EC.

https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170

AIUI our AESNI code will not function properly when built for Arm64EC (and probably won't build). I think we need to do: `((defined(_M_X64) || defined(_M_AMD64)) && !defined(_M_ARM64EC))`